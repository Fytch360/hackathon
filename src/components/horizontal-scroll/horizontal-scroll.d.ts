import {
    HorizontalScrollModeInterface,
    OnClickHTMLButtonElement,
    ScrollContainer
} from '../..'

export type HorizontalScrollMode = keyof HorizontalScrollModeInterface

export interface HorizontalScrollA11y {
    titleBackward?: string;
    titleForward?: string;
}

export type HorizontalScrollOnScroll = (event: React.SyntheticEvent<HTMLDivElement>) => void
export type HorizontalScrollRefWrapper = (element: React.RefObject<HTMLDivElement>) => void


export type HorizontalScrollGetScrollWidth = (scrollLeft: number, current: React.RefObject<HTMLDivElement>) => void

export interface HorizontalScrollProps extends React.HTMLProps<HTMLDivElement> {
    children: React.ReactNode;
    mode?: HorizontalScrollMode;
    onScroll?: HorizontalScrollOnScroll;
    refWrapper?: HorizontalScrollRefWrapper;
    a11y?: HorizontalScrollA11y;
    scrollWidth?: number;
    getScrollWidth?: HorizontalScrollGetScrollWidth;
    scrollContainer?: ScrollContainer;
}

export const HorizontalScroll: React.FC<HorizontalScrollProps>

export type SegmentedHorizontalScrollMode = keyof HorizontalScrollModeInterface

export interface SegmentedHorizontalScrollA11y {
    titleBackward?: string;
    titleForward?: string;
}

export interface SegmentedHorizontalScrollScrollContainer {
    Inner?: React.ComponentType;
    Outer?: React.ComponentType;
}
export type SegmentedHorizontalScrollOnChange = (value: string) => void
export interface SegmentedHorizontalScrollProps {
    children: React.ReactNode;
    mode?: SegmentedHorizontalScrollMode;
    onScroll?: HorizontalScrollOnScroll;
    onChange?: SegmentedHorizontalScrollOnChange;
    a11y?: SegmentedHorizontalScrollA11y;
    scrollWidth?: number;
    getScrollWidth?: HorizontalScrollGetScrollWidth;
    scrollContainer?: SegmentedHorizontalScrollScrollContainer;
    refWrapper?: HorizontalScrollRefWrapper;
}

export const SegmentedHorizontalScroll: React.FC<SegmentedHorizontalScrollProps>

export interface ScrollSegmentProps {
    title?: string;
    onClick?: OnClickHTMLButtonElement;
    active?: boolean;
    disabled?: boolean;
    className?: string;
}

export const ScrollSegment: React.FC<ScrollSegmentProps>
