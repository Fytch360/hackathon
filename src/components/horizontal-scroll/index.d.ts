export * from './horizontal-scroll.d'
