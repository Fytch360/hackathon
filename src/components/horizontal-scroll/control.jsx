import React from 'react'
import PropTypes from 'prop-types'

import { Icon } from '../icon'
import { ic24ArrowLeft, ic24ArrowRight, ic24ChevronRight } from '../icon/common'

import {
    ControlStyled,
    ButtonStyled,
    IconShadowStyled
} from './horizontal-scroll.style'
import scrollShadow from './scroll-shadow.svg'


const modesIcons = {
    block: {
        left: ic24ChevronRight,
        right: ic24ChevronRight
    },
    banner: {
        left: ic24ChevronRight,
        right: ic24ChevronRight
    },
    inline: {
        left: ic24ArrowLeft,
        right: ic24ArrowRight
    }
}

export const Control = ({ side, onClick, title, active, mode }) => (
    <ControlStyled
        mode={mode}
        side={side}
        active={active}
    >
        {mode === 'block' &&
        <IconShadowStyled
            icon={scrollShadow}
            side={side}
            active={active}
        />
        }
        <ButtonStyled
            onClick={onClick}
            disabled={!active}
            type="button"
            title={title}
            mode={mode}
            side={side}
            active={active}
            tabIndex={active ? 0 : -1}
        >
            <Icon icon={modesIcons[mode][side]} />
        </ButtonStyled>
    </ControlStyled>
)

Control.propTypes = {
    onClick: PropTypes.func.isRequired,
    title: PropTypes.string.isRequired,
    active: PropTypes.bool.isRequired,
    side: PropTypes.oneOf(['left', 'right']).isRequired,
    mode: PropTypes.oneOf(['inline', 'block', 'banner']).isRequired
}
