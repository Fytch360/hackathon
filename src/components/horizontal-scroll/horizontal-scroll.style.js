import styled from "@emotion/styled/macro"
import { css } from '@emotion/react'

import { Icon } from '../icon'
import { zIndexTooltip } from '../styles/z-index.config.style'
import { baseX } from '../styles/semantic.config.style'
import { mdShadow, smShadow } from '../styles/shadows.config.style'
import { focusBorder } from '../styles/dynamic-styles'

const SHADOW_OFFSET = 6
const BUTTON_BLOCK_SIZE = 12
const BUTTON_INLINE_SIZE = 6

export const ScrollStyled = styled.div(({ mode }) => css`
    position: relative;

    ${mode === 'inline' && css`
        display: flex;
    `}
`)

export const ScrollZoneStyled = styled.div(({ mobile, mode, leftControlHidden, rightControlHidden }) => css`
    white-space: nowrap;
    display: flex;
    overflow-x: hidden;

    ${mode === 'block' && css`
        margin: -${SHADOW_OFFSET * baseX}px 0;
        padding: ${SHADOW_OFFSET * baseX}px 0;

        ${(leftControlHidden || mobile) && css`
                margin-left: -${SHADOW_OFFSET * baseX}px;

                &::before {
                    content: '';
                    padding-left: ${SHADOW_OFFSET * baseX}px;
                }
            `}

        ${(rightControlHidden || mobile) && css`
                margin-right: -${SHADOW_OFFSET * baseX}px;

                &::after {
                    content: '';
                    padding-right: ${SHADOW_OFFSET * baseX}px;
                }
            `}
    `}

    ${mobile && css`
        overflow-x: auto;
        scroll-behavior: smooth;

        &::-webkit-scrollbar {
            display: none;
        }
    `}
`)

export const ContentsStyled = styled.div`
    display: inline-flex;
    flex-shrink: 0;
`

export const ControlStyled = styled.div(({ mode, side, active, theme }) => css`
    flex-shrink: 0;
    z-index: ${zIndexTooltip - 1};
    transition: width 0.17s ease-in-out;
    position: absolute;
    height: 100%;
    top: 0;
    pointer-events: none;
    display: flex;
    align-items: center;
    ${mode === 'block' && side === 'left' && css`
        left: -${BUTTON_BLOCK_SIZE * baseX / 2}px;

        & svg {
            transform: rotate(180deg);
        }
    `}
    ${mode === 'block' && side === 'right' && css`
        right: -${BUTTON_BLOCK_SIZE * baseX / 2}px;
    `}
    ${mode === 'banner' && side === 'left' && css`
        left: -${BUTTON_BLOCK_SIZE * baseX / 2}px;

        & svg {
            transform: rotate(180deg);
        }
    `}
    ${mode === 'banner' && side === 'right' && css`
        right: -${BUTTON_BLOCK_SIZE * baseX / 2}px;
    `}
    ${mode === 'inline' && side === 'left' && css`
        left: 0;
    `}
    ${mode === 'inline' && side === 'left' && active && css`
        &::after {
            content: '';
            width: 36px;
            height: 100%;
            background-image: linear-gradient(to right, ${theme.horizontalScrollGradientFrom}, ${theme.horizontalScrollGradientTo});
        }
    `}
    ${mode === 'inline' && side === 'right' && css`
        right: 0;
    `}
    ${mode === 'inline' && side === 'right' && active && css`
        &::before {
            content: '';
            width: 36px;
            height: 100%;
            background-image: linear-gradient(to left, ${theme.horizontalScrollGradientFrom}, ${theme.horizontalScrollGradientTo});
        }
    `}
`)

const blockButtonStyle = ({ mode, active, theme }) => (mode === 'block' || mode === 'banner') && css`
    width: ${BUTTON_BLOCK_SIZE * baseX}px;
    height: ${BUTTON_BLOCK_SIZE * baseX}px;
    border: 1px solid ${theme.buttonElevationBoderNormal};
    border-radius: 50%;
    box-shadow: ${smShadow(theme)};
    background-color: ${theme.buttonElevationBody};

    &:hover {
        box-shadow: ${mdShadow(theme)};
        border: 1px solid ${theme.buttonElevationBoderHover};
    }

    ${active && css`
        ${ScrollStyled}:hover & {
            opacity: 1;
        }

        body:not(.pointer-events) &:focus {
            opacity: 1;

            ${focusBorder(theme.focusColor, '1px')};

            border-color: ${theme.focusColor};
        }
    `}
`

const inlineButtonStyle = ({ mode, active, theme }) => mode === 'inline' && css`
    width: ${BUTTON_INLINE_SIZE * baseX}px;
    height: 100%;
    background-color: ${theme.horizontalScrollGradientFrom};

    ${active && css`
        opacity: 1;

        body:not(.pointer-events) &:focus {
            ${focusBorder(theme.focusColor)};
        }
    `}
`

export const ButtonStyled = styled.button(({ theme, active }) => css`
    padding: 0;
    margin: 0;
    opacity: 0;
    outline: none;
    position: relative;
    transition:
        opacity 0.17s ease-in-out,
        box-shadow 0.17s ease-in-out;

    svg {
        transition: fill 0.17s ease-in-out;
        fill: ${theme.buttonElevationText};
    }
    
    ${active && css`
        cursor: pointer;
        pointer-events: visible;
    `}
`,
blockButtonStyle,
inlineButtonStyle
)

export const IconShadowStyled = styled.span(({ side, active }) => css`
    position: absolute;
    top: 0;
    opacity: 0;
    height: 100%;
    transition: opacity 0.17s linear;
    ${side === 'left' && css`
        left: 50%;

        svg {
            transform: rotate(180deg);
        }
    `}
    ${side === 'right' && css`
        right: 50%;
    `}
    ${active && css`
        opacity: 1;
    `}
`).withComponent(Icon)

export const NavigationStyled = styled.div`
    display: flex;
    justify-content: center;
    padding-top: 16px;
    padding-bottom: 10px;
`

export const DotsStyled = styled.button(({ active, theme }) => css`
    margin-right: 6px;

    &:last-child {
        margin-right: 0;
    }

    &:hover {
        transition: background-color 0.37s ease-in-out;
        background-color: ${theme.bannerNavigationButtonHover};
    }

    background-color: ${theme.bannerNavigationButtonNormal};
    transition: all 0.77s ease-in-out;
    width: 6px;
    height: 6px;
    border-radius: 3px;
    overflow: visible;
    padding: 0;
    cursor: pointer;

    ${active && css`
        transition: all 0.77s ease-in-out;
        background-color: ${theme.bannerNavigationButtonClick};
        width: 18px;
    `}
`)
