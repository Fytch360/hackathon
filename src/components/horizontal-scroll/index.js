export { HorizontalScroll } from './horizontal-scroll'
export { SegmentedHorizontalScroll, ScrollSegment } from './segmented-horizontal-scroll'
