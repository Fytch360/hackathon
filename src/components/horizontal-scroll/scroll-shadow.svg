<svg width="16" height="100%" viewBox="0 0 16 192" fill="none" xmlns="http://www.w3.org/2000/svg" preserveAspectRatio="none">
    <mask id="mask0" mask-type="alpha" maskUnits="userSpaceOnUse" x="0" y="0" width="16" height="192">
        <rect width="16" height="192" fill="url(#paint0_linear)"/>
    </mask>
    <g mask="url(#mask0)">
        <rect width="16" height="192" fill="url(#paint1_linear)"/>
    </g>
    <defs>
        <linearGradient id="paint0_linear" x1="13" y1="192" x2="11.4683" y2="0.000249678" gradientUnits="userSpaceOnUse">
            <stop stop-color="white" stop-opacity="0"/>
            <stop offset="0.286458" stop-color="white"/>
            <stop offset="0.666667" stop-color="white"/>
            <stop offset="1" stop-color="white" stop-opacity="0"/>
        </linearGradient>
        <linearGradient id="paint1_linear" x1="16" y1="86.7097" x2="-5.77131e-07" y2="86.7097" gradientUnits="userSpaceOnUse">
            <stop stop-opacity="0.1"/>
            <stop offset="0.489583" stop-color="#C4C4C4" stop-opacity="0"/>
        </linearGradient>
    </defs>
</svg>
