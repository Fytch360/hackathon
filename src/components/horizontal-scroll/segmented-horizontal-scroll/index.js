export { SegmentedHorizontalScroll } from './segmented-horizontal-scroll'
export { ScrollSegment } from './scroll-segment'
