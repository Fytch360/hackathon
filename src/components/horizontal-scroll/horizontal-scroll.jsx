import React, { useState, useEffect, useCallback, useRef } from 'react'
import PropTypes from 'prop-types'

import { isMobilePlatform } from '../utils/adaptive'
import { disableHandler } from '../utils/handlers'

import { leftCheck, rightCheck, smoothScroll } from './utils'
import {
    ScrollStyled,
    ScrollZoneStyled,
    ContentsStyled,
    NavigationStyled,
    DotsStyled
} from './horizontal-scroll.style'
import { Control } from './control'

const SCROLL_WIDTH = 300
const SCROLL_TIME = 1000
// comment: некорректный парсинг jsdoc
/* eslint-disable valid-jsdoc */
/**
 * Технический компонент для управления горизонтальным скроллом в списке карточек
 *
 * @param {Object} props - свойства компонента
 * @return {JSX} - компонент
 */

export const HorizontalScroll = ({
    children,
    onScroll = () => {},
    mode = 'inline',
    refWrapper = () => {},
    a11y: {
        titleBackward = 'Листать назад',
        titleForward = 'Листать вперед'
    } = {},
    parentId,
    scrollWidth = SCROLL_WIDTH,
    getScrollWidth,
    navigation = false,
    scrollContainer: {
        Inner,
        Outer
    } = {}
}) => {
    const mobile = isMobilePlatform()

    const [controlsLocked, setControlsLocked] = useState(false)
    const [activeOffer, setActive] = useState(0)
    const [, setMounted] = useState(false)
    const scroll = useRef(null)
    const [activeState, setState] = useState({
        left: false,
        right: false
    })
    const blocks = children?.props?.children || children

    const checkContent = () => {
        setState({
            left: leftCheck(scroll.current),
            right: rightCheck(scroll.current)
        })
    }

    useEffect(() => {
        setMounted()
        checkContent()

        if (activeOffer === blocks.length && activeOffer > 0) {
            setActive((activeDot) => activeDot - 1)
        }
    }, [activeState.left, activeState.right, activeOffer, blocks.length])

    const takeSteps = useCallback((step) => {
        if (activeOffer + step >= 0 && activeOffer + step !== blocks.length) {
            setActive((activeDot) => activeDot + step)
        }
    }, [activeOffer, blocks.length])

    const handleScrollEnd = useCallback(() => {
        setControlsLocked(false)

        checkContent()
    }, [])

    const nextValue = useCallback((steps) => {
        const { current } = scroll

        return (
            getScrollWidth
                ? steps * getScrollWidth(current.scrollLeft, current)
                : steps * scrollWidth
        )
    }, [])

    const handleLeftClick = useCallback((event) => {
        const id = event.target?.id.split('nav')[1]
        const steps = navigation && id ? activeOffer - id : 1

        setControlsLocked(true)
        
        takeSteps(-steps)
        smoothScroll(scroll.current, -nextValue(steps), SCROLL_TIME, handleScrollEnd)
    }, [activeOffer, nextValue, takeSteps, handleScrollEnd])

    const handleRightClick = useCallback((event) => {
        const id = event.target?.id.split('nav')[1]
        const steps = navigation && id ? id - activeOffer : 1

        setControlsLocked(true)
        
        takeSteps(steps)
        smoothScroll(scroll.current, nextValue(steps), SCROLL_TIME, handleScrollEnd)
    }, [activeOffer, nextValue, takeSteps, handleScrollEnd])

    const OuterScrollStyled = mobile && Outer ? ScrollStyled.withComponent(Outer) : ScrollStyled
    const InnerScrollStyled = mobile && Inner ? ContentsStyled.withComponent(Inner) : ContentsStyled

    return (
        <>
            <OuterScrollStyled
                ref={refWrapper}
                mobile={mobile}
                mode={mode}
            >
                {!mobile && (
                    <Control
                        side="left"
                        onClick={disableHandler(handleLeftClick, controlsLocked)}
                        title={titleBackward}
                        active={activeState.left}
                        mode={mode}
                    />
                )}
                <ScrollZoneStyled
                    id={parentId}
                    ref={scroll}
                    onScroll={onScroll}
                    mobile={mobile}
                    mode={mode}
                    leftControlHidden={!activeState.left}
                    rightControlHidden={!activeState.right}
                    onMouseEnter={disableHandler(checkContent, mobile)}
                >
                    <InnerScrollStyled>
                        {children}
                    </InnerScrollStyled>
                </ScrollZoneStyled>
                {!mobile && (
                    <Control
                        side="right"
                        onClick={disableHandler(handleRightClick, controlsLocked)}
                        title={titleForward}
                        active={activeState.right}
                        mode={mode}
                    />
                )}
            </OuterScrollStyled>
            {!mobile && navigation && Array.isArray(blocks) && blocks.length > 1 &&
            <NavigationStyled>
                {blocks.map(({ key }, index) => (
                    <DotsStyled
                        active={activeOffer === index}
                        aria-hidden
                        id={`${key}-scroll-nav${index}`}
                        key={key}
                        onClick={activeOffer > index ? handleLeftClick : handleRightClick}
                        disabled={controlsLocked}
                    />
                ))}
            </NavigationStyled>}
        </>
    )
}

HorizontalScroll.propTypes = {
    children: PropTypes.node.isRequired,
    mode: PropTypes.oneOf(['inline', 'block', 'banner']),
    onScroll: PropTypes.func,
    refWrapper: PropTypes.func,
    a11y: PropTypes.shape({
        titleBackward: PropTypes.string,
        titleForward: PropTypes.string
    }),
    scrollWidth: PropTypes.number,
    getScrollWidth: PropTypes.func,
    navigation: PropTypes.bool,
    scrollContainer: PropTypes.shape({
        Inner: PropTypes.func,
        Outer: PropTypes.func,
    })
}
