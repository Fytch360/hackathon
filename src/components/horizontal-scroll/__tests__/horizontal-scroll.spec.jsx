import React from 'react'
import { mount } from 'enzyme'
import { matchers } from '@emotion/jest'
import { HorizontalScroll } from '../horizontal-scroll'

expect.extend(matchers)

describe('HorizontalScroll tests', () => {
    test('HorizontalScroll renders correctly', () => {
        const wrapper = mount(<HorizontalScroll>This is HorizontalScroll.</HorizontalScroll>)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper).toHaveStyleRule('display', 'flex')
    })
})
