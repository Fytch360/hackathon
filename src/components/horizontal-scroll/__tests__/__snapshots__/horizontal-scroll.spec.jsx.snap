// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalScroll tests HorizontalScroll renders correctly 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  z-index: 9;
  -webkit-transition: width 0.17s ease-in-out;
  transition: width 0.17s ease-in-out;
  position: absolute;
  height: 100%;
  top: 0;
  pointer-events: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  left: 0;
}

.emotion-4 {
  padding: 0;
  margin: 0;
  opacity: 0;
  outline: none;
  position: relative;
  -webkit-transition: opacity 0.17s ease-in-out,box-shadow 0.17s ease-in-out;
  transition: opacity 0.17s ease-in-out,box-shadow 0.17s ease-in-out;
  width: 24px;
  height: 100%;
}

.emotion-4 svg {
  -webkit-transition: fill 0.17s ease-in-out;
  transition: fill 0.17s ease-in-out;
}

.emotion-6 {
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: middle;
}

.emotion-6 svg {
  display: block;
}

.emotion-8 {
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow-x: hidden;
}

.emotion-10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-12 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  z-index: 9;
  -webkit-transition: width 0.17s ease-in-out;
  transition: width 0.17s ease-in-out;
  position: absolute;
  height: 100%;
  top: 0;
  pointer-events: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  right: 0;
}

<HorizontalScroll>
  <ScrollStyled
    mobile={false}
    mode="inline"
  >
    <div
      className="emotion-0 emotion-1"
      mode="inline"
    >
      <Control
        active={false}
        mode="inline"
        onClick={[Function]}
        side="left"
        title="Листать назад"
      >
        <ControlStyled
          active={false}
          mode="inline"
          side="left"
        >
          <div
            className="emotion-2 emotion-3"
            mode="inline"
          >
            <ButtonStyled
              active={false}
              disabled={true}
              mode="inline"
              onClick={[Function]}
              side="left"
              tabIndex={-1}
              title="Листать назад"
              type="button"
            >
              <button
                className="emotion-4 emotion-5"
                disabled={true}
                mode="inline"
                onClick={[Function]}
                tabIndex={-1}
                title="Листать назад"
                type="button"
              >
                <Icon
                  icon="<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\">    <path fill-rule=\\"evenodd\\" d=\\"M19 11H7.83l4.88-4.88c.39-.39.39-1.03 0-1.42-.39-.39-1.02-.39-1.41 0l-6.59 6.59c-.39.39-.39 1.02 0 1.41l6.59 6.59c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L7.83 13H19c.55 0 1-.45 1-1s-.45-1-1-1z\\"/></svg>"
                >
                  <WrapperStyled
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\">    <path fill-rule=\\"evenodd\\" d=\\"M19 11H7.83l4.88-4.88c.39-.39.39-1.03 0-1.42-.39-.39-1.02-.39-1.41 0l-6.59 6.59c-.39.39-.39 1.02 0 1.41l6.59 6.59c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L7.83 13H19c.55 0 1-.45 1-1s-.45-1-1-1z\\"/></svg>",
                      }
                    }
                    onClick={[Function]}
                  >
                    <span
                      className="emotion-6 emotion-7"
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\">    <path fill-rule=\\"evenodd\\" d=\\"M19 11H7.83l4.88-4.88c.39-.39.39-1.03 0-1.42-.39-.39-1.02-.39-1.41 0l-6.59 6.59c-.39.39-.39 1.02 0 1.41l6.59 6.59c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L7.83 13H19c.55 0 1-.45 1-1s-.45-1-1-1z\\"/></svg>",
                        }
                      }
                      onClick={[Function]}
                    />
                  </WrapperStyled>
                </Icon>
              </button>
            </ButtonStyled>
          </div>
        </ControlStyled>
      </Control>
      <ScrollZoneStyled
        leftControlHidden={true}
        mobile={false}
        mode="inline"
        onMouseEnter={[Function]}
        onScroll={[Function]}
        rightControlHidden={true}
      >
        <div
          className="emotion-8 emotion-9"
          mode="inline"
          onMouseEnter={[Function]}
          onScroll={[Function]}
        >
          <ContentsStyled>
            <div
              className="emotion-10 emotion-11"
            >
              This is HorizontalScroll.
            </div>
          </ContentsStyled>
        </div>
      </ScrollZoneStyled>
      <Control
        active={false}
        mode="inline"
        onClick={[Function]}
        side="right"
        title="Листать вперед"
      >
        <ControlStyled
          active={false}
          mode="inline"
          side="right"
        >
          <div
            className="emotion-12 emotion-3"
            mode="inline"
          >
            <ButtonStyled
              active={false}
              disabled={true}
              mode="inline"
              onClick={[Function]}
              side="right"
              tabIndex={-1}
              title="Листать вперед"
              type="button"
            >
              <button
                className="emotion-4 emotion-5"
                disabled={true}
                mode="inline"
                onClick={[Function]}
                tabIndex={-1}
                title="Листать вперед"
                type="button"
              >
                <Icon
                  icon="<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" >    <path transform=\\"translate(24 24) rotate(180)\\" fill-rule=\\"evenodd\\" d=\\"M19 11H7.83l4.88-4.88c.39-.39.39-1.03 0-1.42-.39-.39-1.02-.39-1.41 0l-6.59 6.59c-.39.39-.39 1.02 0 1.41l6.59 6.59c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L7.83 13H19c.55 0 1-.45 1-1s-.45-1-1-1z\\"/></svg>"
                >
                  <WrapperStyled
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" >    <path transform=\\"translate(24 24) rotate(180)\\" fill-rule=\\"evenodd\\" d=\\"M19 11H7.83l4.88-4.88c.39-.39.39-1.03 0-1.42-.39-.39-1.02-.39-1.41 0l-6.59 6.59c-.39.39-.39 1.02 0 1.41l6.59 6.59c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L7.83 13H19c.55 0 1-.45 1-1s-.45-1-1-1z\\"/></svg>",
                      }
                    }
                    onClick={[Function]}
                  >
                    <span
                      className="emotion-6 emotion-7"
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" >    <path transform=\\"translate(24 24) rotate(180)\\" fill-rule=\\"evenodd\\" d=\\"M19 11H7.83l4.88-4.88c.39-.39.39-1.03 0-1.42-.39-.39-1.02-.39-1.41 0l-6.59 6.59c-.39.39-.39 1.02 0 1.41l6.59 6.59c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L7.83 13H19c.55 0 1-.45 1-1s-.45-1-1-1z\\"/></svg>",
                        }
                      }
                      onClick={[Function]}
                    />
                  </WrapperStyled>
                </Icon>
              </button>
            </ButtonStyled>
          </div>
        </ControlStyled>
      </Control>
    </div>
  </ScrollStyled>
</HorizontalScroll>
`;
